sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -554,8 +554,10 @@ class Analyzer(
           Cast(value, pivotColumn.dataType, Some(conf.sessionLocalTimeZone)).eval(EmptyRow)
         }
         // Group-by expressions coming from SQL are implicit and need to be deduced.
+        val pivotColAndAggRefs =
```

Comment:
`pivotColAndAggRefs` is used inside of `getOrElse` only. Could you move it there.

Reply:
Nice advice!